auth/token_expiration_test.go

```diff
@@ -0,0 +1,304 @@
+package auth_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/NordSecurity/nordvpn-linux/config"
+	"github.com/NordSecurity/nordvpn-linux/core"
+	"github.com/NordSecurity/nordvpn-linux/daemon/response"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"github.com/NordSecurity/nordvpn-linux/request"
+	"github.com/NordSecurity/nordvpn-linux/session"
+	"github.com/NordSecurity/nordvpn-linux/test/category"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	defaultTokenValue     = "valid-token"
+	renewedTokenValue     = "new-token-value"
+	testUsername          = "testuser"
+	testEmail             = "test@example.com"
+	defaultExpirationDays = 24 * time.Hour
+)
+
+type mockConfigManager struct {
+	config    config.Config
+	loadError error
+	saveError error
+}
+
+func (m *mockConfigManager) Load(cfg *config.Config) error {
+	if m.loadError != nil {
+		return m.loadError
+	}
+	*cfg = m.config
+	return nil
+}
+
+func (m *mockConfigManager) SaveWith(f config.SaveFunc) error {
+	if m.saveError != nil {
+		return m.saveError
+	}
+	m.config = f(m.config)
+	return nil
+}
+
+func (m *mockConfigManager) Reset(preserveLoginData bool, disableKillswitch bool) error {
+	return nil
+}
+
+func (m *mockConfigManager) SetTokenExpiry(value time.Time) {
+	dt := m.config.TokensData[m.config.AutoConnectData.ID]
+	dt.TokenExpiry = value.Format(internal.ServerDateFormat)
+	m.config.TokensData[m.config.AutoConnectData.ID] = dt
+}
+
+type mockRoundTripperWithExpiration struct {
+	renewResp              *core.TokenRenewResponse
+	respError              error
+	expectedIdempotencyKey uuid.UUID
+	createdAt              time.Time
+	currentUserResp        *core.CurrentUserResponse
+	RoundTripFunc          func(req *http.Request) (*http.Response, error)
+}
+
+func (rt *mockRoundTripperWithExpiration) RoundTrip(req *http.Request) (*http.Response, error) {
+	return rt.RoundTripFunc(req)
+}
+
+func (rt *mockRoundTripperWithExpiration) ForceExpiration() {
+	rt.createdAt = time.Now().Add(-2 * defaultExpirationDays)
+}
+
+func (rt *mockRoundTripperWithExpiration) ResetExpiration() {
+	rt.createdAt = time.Now()
+}
+
+func createUnauthorizedResponse() (*http.Response, error) {
+	errorBody := `{
+        "errors": {
+            "code": 101301,
+            "message": "Unauthorized"
+        }
+    }`
+
+	return &http.Response{
+		Status:     http.StatusText(http.StatusUnauthorized),
+		StatusCode: http.StatusUnauthorized,
+		Header: http.Header{
+			"Content-Type": []string{"application/json"},
+		},
+		Body:          io.NopCloser(bytes.NewReader([]byte(errorBody))),
+		ContentLength: int64(len(errorBody)),
+	}, nil
+}
+
+func createDefaultRoundTripper(rt *mockRoundTripperWithExpiration) func(req *http.Request) (*http.Response, error) {
```

Comment:
This could be done with `t.Helper()`, then you can fail with `t.Error()` or similar instead of panic and it will show the line in the test that failed